torch/nn/utils/rnn.py

```diff
@@ -82,7 +104,19 @@ def pack_padded_sequence(input, lengths, batch_first=False):
     return PackedSequence(torch.cat(steps), batch_sizes)
 
 
-def pad_packed_sequence(sequence, batch_first=False, padding_value=0.0):
+_zero_by_type = {
+    'torch.DoubleTensor': 0.0,
+    'torch.FloatTensor': 0.0,
+    'torch.HalfTensor': 0.0,
+    'torch.LongTensor': 0,
+    'torch.IntTensor': 0,
+    'torch.ShortTensor': 0,
+    'torch.CharTensor': 0,
+    'torch.ByteTensor': 0,
+}
```

Comment:
Is that really necessary? It seemed to be working fine before. A better way would be to use `torch._integer_tensor_classes` (the map you wrote here doesn't handle e.g. CUDA and sparse types).

Reply:
Yeah, except for int types:
```import torch
import numpy as np
from torch.autograd import Variable
batch_size = 12
max_length = 4
sequences = [
    Variable(torch.rand(np.random.randint(1, max_length+1)))
    for _ in range(batch_size)
]
ordered = sorted(sequences, key=len, reverse=True)
lengths = [len(x) for x in ordered]
padded = torch.nn.utils.rnn.pad_sequence(ordered)
padded = padded.int()
packed = torch.nn.utils.rnn.pack_padded_sequence(padded, lengths)
torch.nn.utils.rnn.pad_packed_sequence(packed)
```

yields `TypeError: fill_ received an invalid combination of arguments - got (float), but expected (int value)` in origin/master and python 3.5.

Reply:
check